File: experiments/issue-302/probe.php
<?php
spl_autoload_register(function ($class) {
    $prefix = 'LinkFoundation\\LinksNotation\\';
    if (str_starts_with($class, $prefix)) {
        require __DIR__ . '/../../php/src/' . str_replace('\\', '/', substr($class, strlen($prefix))) . '.php';
    }
});
$docs = ["# ok line\n# break: two\nci_gate x\n", "a: b: c", "a (b\n", "a b)\n", ":"];
foreach ($docs as $doc) {
    try {
        $links = (new LinkFoundation\LinksNotation\Parser())->parse($doc);
        echo json_encode($doc) . " -> PARSED " . count($links) . " links\n";
    } catch (Throwable $e) {
        echo json_encode($doc) . " -> " . get_class($e) . ": " . $e->getMessage() . "\n";
    }
}

File: experiments/issue-302/probe.py
import sys
sys.path.insert(0, "python")
from links_notation.parser import Parser
docs = ["# ok line\n# break: two\nci_gate x\n", "a: b: c", "a (b\n", "a b)\n", ":"]
for d in docs:
    try:
        r = Parser().parse(d)
        print(f"{d!r} -> PARSED {r}")
    except Exception as e:
        print(f"{d!r} -> {type(e).__name__}: {e}")
